### PR TITLE
KAA-1190 Prevent security key pair generation by SDK

### PR DESCRIPTION
**Pull request by maxml**
http://jira.kaaproject.org/browse/KAA-1190

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

File: server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java [hunk 1]
```diff
@@ -1124,14 +1111,22 @@ public FileData generateSdk(SdkProfileDto sdkProfile, SdkPlatform platform) thro
...
         }
 
+
         LOG.debug("Sdk profile for sdk generation: {}", sdkProfile);
 
         SdkGenerator generator = SdkGeneratorFactory.createSdkGenerator(platform);
+
+        boolean isTrustfulCredentialService = isTrustfulCredentialService(sdkProfile.getApplicationId());
+        if (!(generator instanceof JavaSdkGenerator)) {
+            isTrustfulCredentialService = true; // only for java, APP-1190
```
Comment (krupet) [inline server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java]: Do not leave such comments in code.
We have some commit messages format:
"<JIRA_TIKET_ID>: commit message"
If someone will have some issues with your code he will find all necessary information from ticket using ticket_id from commit message.

File: server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java [hunk 2]
```diff
@@ -254,28 +232,38 @@
...
 
-    /** The resolver. */
+    /**
+     * The resolver.
+     */
     private volatile OperationsServerResolver resolver;
 
-    /** The zk lock. */
+    /**
+     * The zk lock.
```
Comment (rasendubi) [inline server/node/src/main/java/org/kaaproject/kaa/server/control/service/DefaultControlService.java:265]: All these comments are ultimately useless and don't make the code better but rather harm it. (For more reasoning see https://bradt.ca/blog/useless-code-comments/)

File: server/node/src/main/java/org/kaaproject/kaa/server/control/service/sdk/JavaSdkGenerator.java [hunk 3]
```diff
@@ -321,12 +322,17 @@
      * The Constant DEFAULT_USER_VERIFIER_TOKEN_VAR.
      */
     private static final String DEFAULT_USER_VERIFIER_TOKEN_VAR = "\\$\\{default_user_verifier_token\\}";
-    
+
+    /**
+     * The Constant DEFAULT_USER_VERIFIER_TOKEN_VAR.
```
Comment (rasendubi) [inline server/node/src/main/java/org/kaaproject/kaa/server/control/service/sdk/JavaSdkGenerator.java:327]: Copy-paste god has failed you.

**Comment by maxml** [comment]
Find an error, restart progress

**Comment by rasendubi** [comment]
Don't close your PR's, just push to the same branch to update it.

**Pull request closed without merging**